src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = calculateIndex(key);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            do {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    break;
+                } else if (currentNode.next == null) {
+                    currentNode.next = new Node<>(key, value, null);
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            } while (currentNode != null);
+        }
     }
```

Comment:
move `size++` to the end of the method
```suggestion
        int index = calculateIndex(key);
        Node<K, V> currentNode = table[index];
        if (currentNode == null) {
            table[index] = new Node<>(key, value, null);
        } else {
            do {
                if (Objects.equals(currentNode.key, key)) {
                    currentNode.value = value;
                    break;
                } else if (currentNode.next == null) {
                    currentNode.next = new Node<>(key, value, null);
                    break;
                }
                currentNode = currentNode.next;
            } while (currentNode != null);
        }
        size++;
    }
```